### ISSUE TITLE
Exception thrown when RabbitMQ transport is available but not used

### ISSUE THREAD

**Issue by mikeminutillo** [Particular/NServiceBus.RabbitMQ#471]
I have a support case where a customer wants to swap back and forth between Learning transport and RabbitMQ transport. When they have a reference to the RabbitMQ transport but have configured the Learning transport they get an exception at startup:

```
>System.Collections.Generic.KeyNotFoundException: The given key (RabbitMQ.Routing TopologySupportsDelayedDelivery) was not present in the dictionary
```

I believe this is coming from here: https://github.com/Particular/NServiceBus.RabbitMQ/blob/master/src/NServiceBus.RabbitMQ/DelayedDelivery/PreventRoutingMessagesToTimeoutManager.cs#L16

Rather than enabled by default, should this feature be explicitly enabled when the transport is selected?

**Comment by bording** [comment]
This has come up before, and we suggest that people exclude the RabbitMQ assembly from scanning if they are referencing it but not using the transport.

**Comment by alex-training** [comment]
@bording, I would suggest at least updating the documentation regarding this issue.

**Comment by adamralph** [comment]
After discussion, we are planning to fix this in Version 5.1.0. We don't have an ETA for that version yet, but there is a know workaround in the meantime.

Both the [community discussion](https://discuss.particular.net/t/disable-transport-libraries-autio-scanning/349) and the [StackOverflow question](https://stackoverflow.com/questions/48519599/rabbitmq-error-when-setting-up-learning-transport-with-nservicebus) are now pointing back to this issue so it can serve as a reference point until 5.1.0 is released.

/cc @alex-training @mikeminutillo @Scooletz
